src/librustc/ty/mod.rs

```diff
@@ -2693,13 +2677,29 @@ fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
     tcx.hir.span_if_local(def_id).unwrap()
 }
 
+
+
+/// If the given def ID describes an item belonging to a trait,
+/// return the ID of the trait that the trait item belongs to.
+/// Otherwise, return `None`.
+fn trait_of_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Option<DefId> {
```

Comment:
I think we can still make `CrateNum` work. There are a few queries using it now. Like:

https://github.com/cramertj/rust/blob/e349452f5213080f948d45b63ee28539c900df40/src/librustc/ty/maps.rs#L723